app/code/core/Mage/Sales/Model/Observer.php

```diff
@@ -45,24 +45,28 @@ class Mage_Sales_Model_Observer
     public function cleanExpiredQuotes($schedule)
     {
         Mage::dispatchEvent('clear_expired_quotes_before', ['sales_observer' => $this]);
-
         $lifetimes = Mage::getConfig()->getStoresConfigByPath('checkout/cart/delete_quote_after');
-        foreach ($lifetimes as $storeId => $lifetime) {
-            $lifetime *= 86400;
 
-            /** @var Mage_Sales_Model_Resource_Quote_Collection $quotes */
-            $quotes = Mage::getModel('sales/quote')->getCollection();
+        foreach ($lifetimes as $storeId => $day) {
+            $day = (int) $day;
+            $lifetime = 86400 * $day;
 
+            /** @var Mage_Sales_Model_Resource_Quote_Collection $quotes */
+            $quotes = Mage::getResourceModel('sales/quote_collection');
```

Comment:
Is it need to call it inside foreach loop?

Reply:
Yes, to respect the configuration of every store.